share/gpodder/extensions/rockbox_coverart.py

```diff
@@ -37,10 +45,101 @@ def on_episode_synced(self, device, episode):
             # get the file and folder names we need
             episode_folder = os.path.dirname(episode.local_filename(False))
             device_folder = device.get_episode_folder_on_device(episode)
-            episode_art = os.path.join(episode_folder, "folder.jpg")
-            device_art = os.path.join(device_folder, self.config.art_name_on_device)
-            # make sure we have art to copy and it doesn't already exist
-            if os.path.isfile(episode_art) and not os.path.isfile(device_art):
-                logger.info('Syncing cover art for %s', episode.channel.title)
-                # copy and rename art
-                shutil.copy(episode_art, device_art)
+
+            # discover desired filetype
+            device_filetype = self.config.art_name_on_device.split(".")[-1]
+            device_filename = self.config.art_name_on_device.strip(".%s" % (device_filetype))
+
+            # only allow jpeg, jpg, and png - if invalid, default to jpg
+            if device_filetype.lower() != "jpeg" and device_filetype.lower() != "jpg"\
+                    and device_filetype.lower() != "png":
+                device_filetype = "jpg"
+
+            # sanitize for filetype checking - "jpg" will not match "jpeg"
+            if device_filetype.upper() == "JPG":
+                device_match_filetype = "JPEG"
+            else:
+                device_match_filetype = device_filetype.upper()
+
+            # barring any way for gpodder core to tell us the source cover art name,
```

Comment:
I'd rather add a method to get artwork location from the model

Reply:
Hmm, I'll try to figure out how to do that then.

Reply:
Seems like the solution was under my nose the whole time, we can get the cover path and filename from CoverDownloader(), provided we know most of it, minus extension. Seems to work pretty well.